src/java.base/share/classes/java/lang/Class.java

```diff
@@ -812,14 +812,19 @@ public Void run() {
      *
      * <p> There are nine predefined {@code Class} objects to represent
      * the eight primitive types and void.  These are created by the Java
-     * Virtual Machine, and have the same names as the primitive types that
+     * Virtual Machine, and have the same {@linkplain #getName() names} as the primitive types that
      * they represent, namely {@code boolean}, {@code byte},
      * {@code char}, {@code short}, {@code int},
      * {@code long}, {@code float}, and {@code double}.
      *
-     * <p> These objects may only be accessed via the following public static
-     * final variables, and are the only {@code Class} objects for which
-     * this method returns {@code true}.
+     * <p>No other class objects are considered primitive.
+     *
+     * @apiNote
+     * These {@code Class} objects can be accessed via the {@code
```

Comment:
It might be clearer if the API note starts with something like "A Class object representing a primitive type" rather than "These Class objects".